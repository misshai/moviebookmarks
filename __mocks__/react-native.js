
const rn = require('react-native');

jest.mock('Linking', () => {
    return {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        openURL: jest.fn(),
        canOpenURL: jest.fn(),
        getInitialURL: jest.fn(),
    }
});

jest.mock('../src/componentes/FilmForm', () => {
    const RealComponent = require.requireActual('View');
    const React = require('React');
    class FilmForm extends React.Component {
        render() {

            return React.createElement('View', this.props, this.props.children);

        }
    }
    return FilmForm;
});

jest.mock('react-native-collapsible/Accordion', ()=>{
    const RealComponent = require.requireActual('View');
    const React = require('React');
    class Accordion extends React.Component {
        render() {

            return React.createElement('View', this.props, this.props.children);

        }
    }
    return Accordion;
});



jest.unmock('redux-mock-store');
jest.unmock('redux-thunk');
module.exports = rn;