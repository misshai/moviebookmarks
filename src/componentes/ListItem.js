import React, { Component } from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { CardSection } from './common';

class ListItem extends Component {
    onRowPress() {
        Actions.editFilm({film: this.props.film});
    }

    render() {
        const { name ,lastWatchedSeason, lastWatchedSerie, hasWatched} = this.props.film;
        const {container ,titleStyle, lastWatchedStyle } = styles;
        return (
            <TouchableWithoutFeedback onPress={this.onRowPress.bind(this)}>
                <View>
                    <CardSection style={container}>
                        <Text style={titleStyle}>
                            {name}
                        </Text>
                        {this.renderLastWatched()}

                    </CardSection>
                </View>
            </TouchableWithoutFeedback>
        );
    }

    renderLastWatched() {
        const { lastWatchedSeason, lastWatchedSerie, hasWatched} = this.props.film;
        const { lastWatchedStyle } = styles;
        if (hasWatched)
            return (
                <Text style={lastWatchedStyle}>
                    Last watched {lastWatchedSeason}:{lastWatchedSerie}
                </Text>);
        return (<Text style={lastWatchedStyle}>
            Not watched
        </Text>);
    }
}

const styles = {
    titleStyle: {
        fontSize: 18,
        paddingLeft: 15
    },
    lastWatchedStyle: {
        fontSize: 13,
    },
    container: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    }
};

export default ListItem;
