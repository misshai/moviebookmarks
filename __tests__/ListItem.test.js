import 'react-native';
import React from 'react';
import ListItem from '../src/componentes/ListItem';
import renderer from 'react-test-renderer';

test('List Item Form', () => {
    const filmMock ={ name:'' ,lastWatchedSeason:'', lastWatchedSerie:'', hasWatched:''};
    const component = renderer.create(
        <ListItem film = {filmMock}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});