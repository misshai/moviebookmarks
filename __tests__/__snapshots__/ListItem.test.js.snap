// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Form 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingLeft": 15,
        }
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Not watched
    </Text>
  </View>
</View>
`;
